src/main/java/com/google/devtools/build/lib/analysis/skylark/SkylarkActionFactory.java

```diff
@@ -199,7 +199,6 @@ public void run(
     buildCommandLine(builder, argumentsList);
     if (executableUnchecked instanceof Artifact) {
       Artifact executable = (Artifact) executableUnchecked;
-      builder.addInput(executable);
```

Comment:
What happens when we have a files to run provider? Will the artifact be included? Is it needed?

Reply:
The spawn builder adds `FilesToRunProvider.getFilesToRun()` to tools. A `FilesToRunProvider` that doesn't include the actual executable seems rather defective.

Reply:
Is there a way to construct such a defective provider from Skylark? If so then this would be an incompatible change.

Reply:
It's possible to get a `FilesToRunProvider` that doesn't directly include the `executable` in `getFilesToRun`. However, `FilesToRunProvider` will always include the executable's runfiles. And, I don't think you can escape transitively depending on the executable via runfiles. Runfiles are forced to contain executable by `RunfilesSupport.create`.

Reply:
OK, let's try it and see what happens. I agree it seems like you've thought through all cases.